exploration/code-mode-introducer.md

```diff
@@ -0,0 +1,224 @@
+# Design Proposal: Choosing a Code Mode Introducer
+
+Status: **Proposed**
+
+<details>
+	<summary>Metadata</summary>
+	<dl>
+		<dt>Contributors</dt>
+		<dd>@aphillips</dd>
+		<dt>First proposed</dt>
+		<dd>2023-11-10</dd>
+		<dt>Pull Requests</dt>
+		<dd>#000</dd>
+	</dl>
+</details>
+
+## Objective
+
+_What is this proposal trying to achieve?_
+
+It must be possible to reliably parse messages.
```

Comment:
"Reliable" is vague. Can this be framed as a property of the grammar? (For example: "The grammar for messages must be LL(k), so that it can be parsed without arbitrary lookahead.")

Reply:
I will clarify. This has to do with determining whether a message is simple or complex (and only that). Any given message must produce a consistent result for this.